apps/src/p5lab/spritelab/commands/criterionCommands.js

```diff
@@ -80,49 +80,40 @@ export const commands = {
   // current or previous frame.
   anySpeechIncludesValues(currentVariables, previousVariables) {
     const spriteIds = this.getSpriteIdsInUse();
-    let result = false;
-    const values = previousVariables
-      ? Object.values(currentVariables).concat(Object.values(previousVariables))
-      : Object.values(currentVariables);
-    for (let i = 0; i < spriteIds.length; i++) {
-      values.forEach(value => {
-        const speechText = this.getLastSpeechBubbleForSpriteId(
-          spriteIds[i]
-        )?.text;
-        const type = typeof speechText;
-        // We only want to set result to true here, so that any positive test
-        // allows the overall criterion to pass.
-        switch (type) {
-          case 'string':
-            if (speechText.includes(value)) {
-              result = true;
-            }
-            break;
-          case 'number':
-            if (speechText === value) {
-              result = true;
-            }
-            break;
-          default:
-            break;
+    const values = Object.values(currentVariables);
+    if (previousVariables) {
+      values.push(...Object.values(previousVariables));
+    }
+
+    for (const spriteId of spriteIds) {
+      const speechText = this.getLastSpeechBubbleForSpriteId(spriteId)?.text;
+      if (speechText) {
+        for (const value of values) {
+          if (typeof value === 'string' && speechText.includes(value)) {
```

Comment:
previously we were checking the type of `speechText` but now we are testing the type of `value`, is this intentional?

Reply:
Great catch! This is a mistake on my part. It's probably more important to check the type of `speechText` since we're assuming it has an `.includes()` method. Usually these types will match (and they did when I tested it), but it's possible they won't. Will fix!